hpat/distributed_api.py

```diff
@@ -330,30 +330,46 @@ def const_slice_getitem(arr, slice_index, start, count):
 
 @overload(const_slice_getitem)
 def const_slice_getitem_overload(arr, slice_index, start, count):
+    '''Provides parallel implementation of getting a const slice from arrays of different types
+
+    Arguments:
+    arr -- part of the input array processed by this processor
+    slice_index -- start and stop of the slice in the input array (same on all ranks)
+    start -- position of first arr element in the input array 
+    count -- lenght of the part of the array processed by this processor
+
+    Return value:
+    Function providing implementation basing on arr type. The function should implement
+    logic of fetching const slice from the array distributed over multiple processes.
+    '''
+
+    # TODO: should this also handle slices not staring from zero?
     if arr == string_array_type:
         reduce_op = Reduce_Type.Sum.value
 
         def getitem_str_impl(arr, slice_index, start, count):
             rank = hpat.distributed_api.get_rank()
             k = slice_index.stop
+
             # get total characters for allocation
             n_chars = np.uint64(0)
-            if k > count:
+            if k > start:
```

Comment:
Looks like the code from if-statement is also valid for else-statement. I mean we can avoid if-else branching here and leave only code from the if-statement.

Reply:
@densmirn That's true, thank you. But I think keeping if statement simplifies code. I will remove the redundant "max(k - start, 0)" in the first branch though, i.e. will make it this:

            if k > start:
                # if slice end is beyond the start of this subset we have to send our elements
                my_end = min(count, k - start)
                my_arr = arr[:my_end]
            else:
                my_arr = arr[:0]